react/src/components/HeaderMenu/HeaderMenuContainer.tsx

```diff
@@ -102,11 +103,32 @@ const HeaderMenuContainer = () => {
     [baseId],
   );
 
+  const authorizedMenuItems: MenuItemsGroupData[] = useMemo(() => {
+    return menuItems.filter((menuItem) => {
+      // If no minimum beta requirement exists for the menu item, it should be included.
+      if (!menuItem.minBeta) {
+        return true;
+      }
+
+      // Ensure that auth0.user is defined and parse the beta_user value.
+      if (auth0.user) {
```

Comment:
This is the same logic as line 11ff. in the useAuthorization hook, can it be combined somehow?